### ISSUE TITLE
[Bug]: Intermittent alarm state update issue

### ISSUE THREAD

**Issue by bogar** [bvis/aegis-hass#270]
### Describe the bug

  I reproduced an intermittent alarm state update issue and captured debug logs.

  Sequence on June 7, 2026:
  1. Night arm -> reflected correctly in Home Assistant
  2. Disarm -> reflected correctly in Home Assistant
  3. Night arm again immediately after -> Ajax app changed, but Home Assistant did not reflect the new armed state

### Steps to reproduce

 Relevant sanitized log snippet:

  ```text
  2026-06-07 21:39:42.499 DEBUG [custom_components.aegis_ajax.api.hts.client] RECV: type=0x08 seq=<redacted> sender=<hub_redacted> link=91 payload=50b
  2026-06-07 21:39:42.500 DEBUG [custom_components.aegis_ajax.coordinator] Space HTS event for space <space_redacted> (byte 0x00): requesting authoritative refresh
  2026-06-07 21:39:42.700 DEBUG [custom_components.aegis_ajax.coordinator] Finished fetching aegis_ajax data in 0.200 seconds (success: True)

  2026-06-07 21:39:47.393 DEBUG [custom_components.aegis_ajax.api.hts.client] RECV: type=0x08 seq=<redacted> sender=<hub_redacted> link=246 payload=50b
  2026-06-07 21:39:47.394 DEBUG [custom_components.aegis_ajax.coordinator] Space HTS event for space <space_redacted> (byte 0x02): requesting authoritative refresh

  2026-06-07 21:40:10.334 DEBUG [custom_components.aegis_ajax.api.hts.client] RECV: type=0x08 seq=<redacted> sender=<hub_redacted> link=220 payload=50b
  2026-06-07 21:40:10.335 DEBUG [custom_components.aegis_ajax.coordinator] Space HTS event for space <space_redacted> (byte 0x00): requesting authoritative refresh
  2026-06-07 21:40:10.545 DEBUG [custom_components.aegis_ajax.coordinator] Finished fetching aegis_ajax data in 0.210 seconds (success: True)

  2026-06-07 21:40:13.213 DEBUG [custom_components.aegis_ajax.api.hts.client] Sent REQUEST_FULL_STATUS to <hub_redacted>
  2026-06-07 21:40:13.315 DEBUG [custom_components.aegis_ajax.api.hts.client] STATUS_BODY non-hub devices (#123 probe): <devices_redacted>
  2026-06-07 21:40:13.316 DEBUG [custom_components.aegis_ajax.coordinator] Manually updated aegis_ajax data
  ```
  So from the logs it looks like the integration does receive the space HTS events and does trigger/complete authoritative refreshes successfully, but the alarm
  control panel state still fails to update on the second rapid transition.

### Expected behavior

The alarm_control_panel state should reflect the second rapid transition to night armed after the refresh completes.

### Home Assistant version

2026.6.1

### Integration version

1.11.0

### Installation method

HACS

### App label (co-branded app)

Ajax

### FCM push notifications configured?

Yes

### Relevant logs

```text
2026-06-07 21:39:42.499 DEBUG [custom_components.aegis_ajax.api.hts.client] RECV: type=0x08 seq=<redacted> sender=<hub_redacted> link=91 payload=50b
2026-06-07 21:39:42.500 DEBUG [custom_components.aegis_ajax.coordinator] Space HTS event for space <space_redacted> (byte 0x00): requesting authoritative refresh
2026-06-07 21:39:42.700 DEBUG [custom_components.aegis_ajax.coordinator] Finished fetching aegis_ajax data in 0.200 seconds (success: True)

2026-06-07 21:39:47.393 DEBUG [custom_components.aegis_ajax.api.hts.client] RECV: type=0x08 seq=<redacted> sender=<hub_redacted> link=246 payload=50b
2026-06-07 21:39:47.394 DEBUG [custom_components.aegis_ajax.coordinator] Space HTS event for space <space_redacted> (byte 0x02): requesting authoritative refresh

2026-06-07 21:40:10.334 DEBUG [custom_components.aegis_ajax.api.hts.client] RECV: type=0x08 seq=<redacted> sender=<hub_redacted> link=220 payload=50b
2026-06-07 21:40:10.335 DEBUG [custom_components.aegis_ajax.coordinator] Space HTS event for space <space_redacted> (byte 0x00): requesting authoritative refresh
2026-06-07 21:40:10.545 DEBUG [custom_components.aegis_ajax.coordinator] Finished fetching aegis_ajax data in 0.210 seconds (success: True)

2026-06-07 21:40:13.213 DEBUG [custom_components.aegis_ajax.api.hts.client] Sent REQUEST_FULL_STATUS to <hub_redacted>
2026-06-07 21:40:13.315 DEBUG [custom_components.aegis_ajax.api.hts.client] STATUS_BODY non-hub devices (#123 probe): <devices_redacted>
2026-06-07 21:40:13.316 DEBUG [custom_components.aegis_ajax.coordinator] Manually updated aegis_ajax data
```

**Comment by bogar** [comment]
I retried the same rapid sequence again and could not reproduce it consistently this time.

  I repeated the transitions several times and in this run it updated correctly, so the issue does appear intermittent.

  What I observed in the successful run is that the HTS/event/refresh pattern looks very similar to the failing run:
  - `EVENT(0x08)` space events are received
  - the integration requests an authoritative refresh
  - the fetch completes successfully
  - `STATUS_UPDATE push` messages follow immediately after

  So at least from the debug output I have, the difference between a good run and a bad run is not an obvious transport/connectivity problem. It looks more like an
  intermittent state propagation/update issue after otherwise normal HTS handling.

  A representative sanitized pattern from the successful run was:

  ```text
  ... Space HTS event for space <space_redacted> (byte 0x02): requesting authoritative refresh ...
  ... Finished fetching aegis_ajax data in 0.250 seconds (success: True)
  ... STATUS_UPDATE push (#123): <device_redacted>=[...]
  ...
  ... Space HTS event for space <space_redacted> (byte 0x00): requesting authoritative refresh ...
  ... Finished fetching aegis_ajax data in 0.197 seconds (success: True)
  ... STATUS_UPDATE push (#123): <device_redacted>=[...]
```
  So for now I would describe it as:

  - intermittent
  - reproducible at least once on a rapid consecutive arm/disarm/arm-night sequence
  - not obviously explained by missing HTS events, failed refreshes, or connection loss in the captured logs

**Comment by bvis** [comment]
A fix is ready in **1.11.1-beta.1**.

**What was happening:** when the panel had to fall back on the hub's arm/disarm event (because an FCM push didn't arrive promptly), the re-read of the real state went through Home Assistant's shared refresh debouncer, whose 10-second cooldown collapsed a quick arm→disarm→arm sequence into a single delayed re-read — so the panel could stay on the previous state for up to ~10 seconds (until the next event or the periodic poll). It only surfaced when push didn't deliver first, which is why it was intermittent. The event-triggered re-read now uses a dedicated ~1-second path, so each transition is picked up within about a second.

Could you update and retry the same rapid sequence?

1. HACS → Aegis for Ajax → ⋮ → **Redownload** → enable **Show beta versions** → pick **1.11.1-beta.1**.
2. Restart Home Assistant.
3. Repeat the night-arm → disarm → night-arm sequence a few times (with and without waiting for the app to settle) and confirm the panel follows each transition.

If it still lags on any transition, please grab the debug log around it again — thanks for the detailed captures so far.

**Comment by bogar** [comment]
I updated to `1.11.1-beta.1`, restarted Home Assistant, and retried the rapid sequence multiple times.

  So far I have not been able to reproduce the issue again.

  Tested on:
  - Home Assistant `2026.6.1`
  - Aegis for Ajax `1.11.1-beta.1`

  I repeated rapid night-arm / disarm / night-arm transitions several times, including consecutive changes without waiting long between them, and the panel followed
  correctly each time in HA.

  From the debug logs, the post-fix behavior looks consistently healthy on each transition:

  - `EVENT(0x08)` is received from HTS
  - the integration immediately requests an authoritative refresh
  - the refresh completes successfully in about `0.14s` to `0.21s`
  - `STATUS_UPDATE push` follows right after
  - I did not see any `ACK failed`, `connection lost`, or delayed ~10s re-read behavior in these runs

  Representative sanitized snippets:

  ```text
  2026-06-08 16:11:14.186 DEBUG [custom_components.aegis_ajax.coordinator] Space HTS event for space <space_redacted> (byte 0x00): requesting authoritative refresh
  2026-06-08 16:11:14.788 DEBUG [custom_components.aegis_ajax.api.hts.client] Hub <hub_redacted>: STATUS_UPDATE push (#123): <device_redacted>=[0x40=27,0xc3=00]
  2026-06-08 16:11:15.345 DEBUG [custom_components.aegis_ajax.coordinator] Finished fetching aegis_ajax data in 0.158 seconds (success: True)

  2026-06-08 16:11:17.206 DEBUG [custom_components.aegis_ajax.coordinator] Space HTS event for space <space_redacted> (byte 0x02): requesting authoritative refresh
  2026-06-08 16:11:17.682 DEBUG [custom_components.aegis_ajax.api.hts.client] Hub <hub_redacted>: STATUS_UPDATE push (#123): <device_redacted>=[0x40=30,0xc3=00]
  2026-06-08 16:11:17.712 DEBUG [custom_components.aegis_ajax.coordinator] Finished fetching aegis_ajax data in 0.210 seconds (success: True)

  2026-06-08 16:11:22.224 DEBUG [custom_components.aegis_ajax.coordinator] Space HTS event for space <space_redacted> (byte 0x00): requesting authoritative refresh
  2026-06-08 16:11:22.839 DEBUG [custom_components.aegis_ajax.api.hts.client] Hub <hub_redacted>: STATUS_UPDATE push (#123): <device_redacted>=[0x40=31,0xc3=00]
  2026-06-08 16:11:23.379 DEBUG [custom_components.aegis_ajax.coordinator] Finished fetching aegis_ajax data in 0.154 seconds (success: True)

  2026-06-08 16:11:25.754 DEBUG [custom_components.aegis_ajax.coordinator] Space HTS event for space <space_redacted> (byte 0x02): requesting authoritative refresh
  2026-06-08 16:11:25.759 DEBUG [custom_components.aegis_ajax.api.hts.client] Hub <hub_redacted>: STATUS_UPDATE push (#123): <device_redacted>=[0x40=32,0xc3=00]
  2026-06-08 16:11:26.304 DEBUG [custom_components.aegis_ajax.coordinator] Finished fetching aegis_ajax data in 0.152 seconds (success: True)
 ```
  Technical observation:

  - in the failing version, the issue looked like a stale panel state after a rapid consecutive transition
  - in this beta, each EVENT(0x08) seems to trigger its own prompt refresh path, and I’m not seeing evidence of transitions being collapsed into a delayed shared re-
    read

  - at least on my setup, this matches the expected effect of the fix

  I’ll keep an eye on it, but for now this looks fixed on my setup.

**Comment by bvis** [comment]
Perfect — thanks for the thorough retest, @bogar. Your logs show exactly the intended behavior: each `EVENT(0x08)` now gets its own prompt re-read (~0.15 s) instead of being collapsed into the shared 10 s re-read.

This is now released in **1.11.1** (stable). Closing as fixed — please reopen if it ever resurfaces.
